### PR TITLE
MP-70 - User profile nudges & toasts

### PR DESCRIPTION
**Pull request by vas3a**
https://topcoder.atlassian.net/browse/MP-70 - profile nudge & completeness toasts

- Implements toasts to encourage users to complete they're profiles
- adds nudge on user profile icon that animates and shows a dot to indicate the "profile" is missing data
- frequency of showing nudge & toasts is handled through cookies
- the toasts are treated as separate logic and is loaded in a separate file in order to not delay the load of the uninav
- animations/assets are lazy loaded as needed (not bundled together with the JS)

### CODE REVIEW

File: src/lib/nudge-app/config.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    ctaText: 'Add Skills',
+    ctaLink: `${PROFILE_HOST}?edit-mode=skills`,
+  },
+  verified: {
+    theme: 'verified',
+    icon: 'verified',
+    cover: 'verified.png',
+    type: 'animated',
+    title: 'Are you real?',
+    message: 'Prove you not a robot by verifying a few details.',
```
Comment (testflyjets) [inline src/lib/nudge-app/config.ts:72]: Should be "Prove you **are** not a robot..."

File: src/lib/nudge-app/config.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ctaText: 'Verify Me',
+    ctaLink: `${PROFILE_HOST}?edit-mode=verified`,
+  },
+  workHistory: {
+    theme: 'workHistory',
+    icon: 'work-history',
+    cover: 'work-history.png',
+    type: 'animated',
+    title: 'Experience says it all!',
+    message: 'Show off your all your hard work and amazing gigs.',
```
Comment (testflyjets) [inline src/lib/nudge-app/config.ts:82]: Should be "Show off ~~your~~ all your hard work..."

File: src/lib/nudge-app/toast-manager.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * @returns
+ */
+export const getToast = (completednessData: ProfileCompletionData) => {
+  if (!completednessData || completednessData.completed || isToastDismissed()) {
+    return;
+  }
+
+  const lastToastSeen = getLastSeenToast();
+  if (!lastToastSeen) {
+    setCookie(TOAST_COOKIE, completednessData.showToast, 7);
```
Comment (testflyjets) [inline src/lib/nudge-app/toast-manager.ts:31]: Is 7 the number of days between showing toasts? Let's make this a variable so it's clear.

**Review by testflyjets — CHANGES_REQUESTED** [review]
Some minor grammar fixes and variable naming. Also, where and how is % complete computed -- by the API?

**Comment by vas3a** [comment]
@testflyjets I addressed the issues above.
The `% complete` is computed on the api side.

**Review by testflyjets — APPROVED** [review]

**Pull request merged**